profiler.sh

```diff
@@ -199,10 +201,21 @@ while [ $# -gt 0 ]; do
             # A shortcut for getting PID of a running Java application
             # -XX:+PerfDisableSharedMem prevents jps from appearing in its own list
             PID=$(pgrep -n java || jps -q -J-XX:+PerfDisableSharedMem)
+            if [ "$PID" = "" ]; then
+                echo "No Java process could be found!"
+            fi
             ;;
         *)
-            echo "Unrecognized option: $1"
-            usage
+            if [ $# -eq 1 ]; then
+                # the last argument is the application name as it would appear in the jps tool
+                PID=$(jps -J-XX:+PerfDisableSharedMem | grep " $1$" | head -n 1 | cut -d ' ' -f 1)
+                if [ "$PID" = "" ]; then
+                    echo "No Java process '$1' could be found!"
+                fi
+            else
+                echo "Unrecognized option: $1"
+                usage
+            fi
```

Comment:
Interesting suggestion! Looks useful.
I'd only exclude options starting with `-`
If there is an unrecognized option starting with `-`, immediately fail without looking for a process

Reply:
I just added a `-*)` case. I would have normally done something bash-y with the `&& [[ ]]` but I saw https://github.com/jvm-profiling-tools/async-profiler/pull/303 was done. I believe what I have is still POSIX compliant since it passes on shellcheck.net

Since this was done